### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by emilie-blackbird**

### CODE REVIEW

File: src/Controller/SearchController.php [hunk 1]
```diff
@@ -92,12 +98,25 @@ public function searchAction(Request $request): Response
             }
         }
 
+        //Prepare pager
+        $pager = $this->preparePager($page, $searchResults['total']);
+
+        //Prepare page help with
+        $pagingHelp = [
+            'start' => ($page - 1) * self::MAX_DISPLAYED_ITEMS_PER_PAGE + 1,
```
Comment (jacquesbh) [inline src/Controller/SearchController.php]: Why do you apply a `+ 1`?
I may be wrong but I think that by doing this you're not getting the first result.

File: src/Controller/SearchController.php [hunk 2]
```diff
@@ -76,8 +81,9 @@ public function postAction(Request $request)
     public function searchAction(Request $request): Response
     {
         $query = htmlspecialchars(urldecode($request->get('query')));
+        $page = (int) htmlspecialchars(urldecode($request->get('page')));
```
Comment (jacquesbh) [inline src/Controller/SearchController.php]: What about `$page <= 0`?

File: src/Controller/SearchController.php [hunk 3]
```diff
@@ -117,9 +136,37 @@ public function instantAction(Request $request): Response
...
+            }elseif($page < $half){
+                $start = 1;
+                $end = self::DIPLAYED_PAGES_PAGER;
+            }elseif($page > $nbTotalPages - $half){
+                $end = $nbTotalPages;
+                $start = $nbTotalPages - self::DIPLAYED_PAGES_PAGER + 1;
+            }
+            return \range($start, $end);
+        }
+    }
```
Comment (jacquesbh) [inline src/Controller/SearchController.php]: Do you think you can review this code for it to be more PSR compliant?

File: src/Indexer/DocumentIndexer.php [hunk 4]
```diff
@@ -262,17 +267,22 @@ private function getIndexer(): Indexer
...
+    private function getSearch($query = null, int $page = 0, $max)
     {
         $elasticJson = $this->searchRequestProvider->getSearchJson();
 
         if ($query) {
             $elasticJson = str_replace('{{QUERY}}', $query, $elasticJson);
         }
 
+        if ($page) {
+            $elasticJson = str_replace('{{FROM}}', $page * $max - $max, $elasticJson);
```
Comment (jacquesbh) [inline src/Indexer/DocumentIndexer.php]: If the `{{FROM}}` isn't replaced, what will happen?

File: src/Resources/config/routing/shop.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 monsieurbiz_sylius_search_search:
-    path: /search/{query}
+    path: /search/{query}/{page<\d+>?1}
```
Comment (jacquesbh) [inline src/Resources/config/routing/shop.yaml]: The whole `/{page<\d+>?1}` should be optional.

File: src/Resources/views/Search/_paging.html.twig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<span class="paging-help">{{ pagingHelp.start }}-{{ pagingHelp.end }} sur {{ pagingHelp.total }}</span>
```
Comment (jacquesbh) [inline src/Resources/views/Search/_paging.html.twig]: Could you add a translation somewhere?

You can use placeholders in your translation, like `%start%` and `%end%`…

File: src/Indexer/DocumentIndexer.php [hunk 7]
```diff
@@ -262,17 +267,22 @@ private function getIndexer(): Indexer
      * Retrieve the JSON to send to Elasticsearch for search
      *
      * @param null $query
+     * @param int $page
      * @return mixed|string
      * @throws ReadFileException
      */
-    private function getSearch($query = null)
+    private function getSearch($query = null, int $page = 0, $max)
```
Comment (jacquesbh) [inline src/Indexer/DocumentIndexer.php]: This is not possible to have the `$max` without default value if the previous argument has one.

File: src/Indexer/DocumentIndexer.php [hunk 8]
```diff
@@ -183,23 +183,28 @@ public function removeOneByLocale(DocumentResult $document, string $locale): voi
      * @param string $locale
      * @param string $query
      * @param int $maxItems
+     * @param int $page
      * @return array
      */
-    public function search(string $locale, string $query, int $maxItems): array
```
Comment (jacquesbh) [inline src/Indexer/DocumentIndexer.php]: Do you have any reason for which you removed the `: array`?

**Comment by maximehuran** [comment]
Replaced by  #14

**Pull request closed without merging**